### PR TITLE
Add new git orb with `custum_checkout` command

### PR DESCRIPTION
**Pull request by cansjt**
Rational is in the orb's description.

### CODE REVIEW

File: orbs/git/orb.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    your repository, but don't really get any history;
+  - it messes with branch references;
+
+  Those are not an issue, until your CI workflow has reasons
+  to switch branches or look at the code's history.
+
+  If you do any of those things, you can use this alternate
+  `custom_checkout` command.
+
+  Of course, if CircleCI does things the way it does, is for
```
Comment (happypoulp) [inline orbs/git/orb.yml:18]: Just a small wording:
```suggestion
  Of course, if CircleCI does things the way it does, it is for
```

File: orbs/git/orb.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          then
+            export HOME=$(getent passwd $(id -un) | cut -d: -f6)
+          fi
+
+          mkdir -p ~/.ssh
+
+          # yamllint disable-line rule:line-length
+          echo 'github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==' >> ~/.ssh/known_hosts
+          # yamllint disable-line rule:line-length
+          echo 'bitbucket.org ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAubiN81eDcafrgMeLzaFPsw2kNvEcqTKl/VqLat/MaB33pZy0y3rJZtnqwR2qOOvbwKZYKiEO1O6VqNEBxKvJJelCq0dTXWT5pbO2gDXC6h6QDXCaHo6pOHGPUy+YBaGQRGuSusMEASYiWunYN0vCAI8QaXnWMXNMdFP3jHAJH0eDsoiGnLPBlBp4TNm6rYI74nMzgz3B9IikW4WVK+dc8KZJZWYjAuORU3jc1c/NPskD2ASinf8v3xnfXeukU0sJ5N6m5E8VLjObPEO+mN2t/FZTMZLiFqPWc/ALSqnMnnhwrNi2rbfg/rd/IpL8Le3pSBne8+seeFVBoGqzHM9yXw==' >> ~/.ssh/known_hosts
```
Comment (happypoulp) [inline orbs/git/orb.yml:44]: **question**

Is it ok to hardcode those values?

I recently saw a similar use case in a github action and I think the recommendation was to use `ssh-keyscan`.

A bit like this:
```
ssh-keyscan github.com >> ~/.ssh/known_hosts
ssh-keyscan bitbucket.org >> ~/.ssh/known_hosts
```

**Review by happypoulp — APPROVED** [review]
👍  LGTM (just one wording and one question about `known_hosts`)

File: orbs/git/orb.yml (hunk 2 above)
Comment (cansjt) [reply orbs/git/orb.yml:44]: This is the way CircleCI does it. This is mostly a modified version of their `checkout` commands.
Security-wise there is no issue those are public keys.

If github or bitbucket changed there public keys we'll know it fast i guess. And yes we'd have to update those values. But they don't change that often. And it is better to have to change them once here than let people pass parameters or whatever and have to change many configurations files.

Whether to use `ssh-keyscan` depends on is it installed in all docker images ? I doubt so.

**Comment by cansjt** [comment]
FYI I don't have the permission required to create an orb in the jobteaser namespace. Hence the publication failure.
```
$ circleci orb create jobteaser/git
You are running 0.1.12331
A new release is available (0.1.15085)
You can update with `circleci update install`

You are creating an orb called "jobteaser/git".

You will not be able to change the name of this orb.

If you change your mind about the name, you will have to create a new orb with the new name.

? Are you sure you wish to create the orb: `jobteaser/git` Yes
Error: AUTHORIZATION_FAILURE
```

**Review by papa-cool — APPROVED** [review]

**Pull request merged**